tests/pmap_test.py

```diff
@@ -115,20 +115,6 @@ class PythonPmapTest(jtu.JaxTestCase):
   def pmap(self):
     return src_api._python_pmap
 
-  def testDeviceBufferToArray(self):
-    sda = self.pmap(lambda x: x)(jnp.ones((jax.device_count(), 2)))
-    buf = sda.device_buffers[-1]
-
-    view = jnp.array(buf, copy=False)
-    self.assertArraysEqual(sda[-1], view)
-    self.assertEqual(buf.device(), view.device())
-    self.assertEqual(buf.unsafe_buffer_pointer(), view.unsafe_buffer_pointer())
-
-    copy = jnp.array(buf, copy=True)
-    self.assertArraysEqual(sda[-1], copy)
-    self.assertEqual(buf.device(), copy.device())
-    self.assertNotEqual(buf.unsafe_buffer_pointer(), copy.unsafe_buffer_pointer())
-
```

Comment:
Why not keep this test as a regression check? If it fails it still indicates a problem with the implementation, I think.

Reply:
I don't think we want to preserve this behavior which relies on `buf = sda.device_buffers[-1]`. That is, we don't want the objects returned by `sda.device_buffers[-1]` to be compatible with `jnp.array`, as we don't want people accessing `device_buffers` at all. (IIRC we used to return objects with `aval=None` from expressions like `sda[-1]`, and that was the problem, but that's now been fixed.)

WDYT? I could adapt the test to set `buf = sda[-1]` rather than `buf.device_buffers[-1]`.

Reply:
Yes, I think that sounds good.